### DISCUSSION TITLE
Pagenotes old sync client deprecation

### DISCUSSION THREAD

**Discussion by manugarg** [manugarg/pagenotes-issues#43] [Announcements]
## What
Pagenotes extension use an API Client to access Google Drive APIs. We've switched to a new API client for a long time now as Google doesn't support the old client anymore.

## Why
Old client continues to work for the existing users (at least for now), but it's highly recommended that you move to the new client, for the following reasons:

 * Newer client uses a safer authentication flow. 
 * Old client works for now, but adding a new device with the old client will most likely not work.
 * __Pagenotes app ([pagenotes.app](https://pagenotes.app))__, a web app to access notes from any device and anywhere, works only with the new client.

## How
1. Go to the extension's options page.
2. Export notes to a CSV file from the options page. (Backup)
4. Click on 'Stop Sync'. This will clear your existing sync setup.
5. Click on 'Setup sync'. New setup will automatically use the new client.

**Comment by drveresh** [comment]
@manugarg  The sync is failing consistently! I tried all (those steps, import, stop/start, etc.) options and nothing worked, but seeing the same error "_There was an error during sync. Click on "Show debug info_" for more information.".  Even its web app shows nothing.

FYI, the Debug error:
Messages from last sync: 
sync: Starting sync at: Wed Jul 26 2023 18:33:05 GMT+0530 (India Standard Time)
sync: First sync, merging local and remote data.
There was a problem downloading the doc. Last request status: 404
```
{
  "error": {
    "code": 404,
    "message": "File not found: 1gF6HFSatQobLUTYxW5_Ais6TRjpxjX5H",
    "errors": [
      {
        "message": "File not found: 1gF6HFSatQobLUTYxW5_Ais6TRjpxjX5H",
        "domain": "global",
        "reason": "notFound",
        "location": "file",
        "locationType": "other"
      }
    ]
  }
}
```

**Comment by manugarg** [comment]
@drveresh Thanks a lot for the feedback! You're encountering this error only after stopping and re-setting up the sync? 

By symptoms, it looks like your existing setup is not clearing up properly. It's trying to use the old pagenotes.data file and unable to find it. Did you delete any file in Google Drive manually?

Stopping and re-setting up the sync will create a new `pagenotes.data` file in your 'Google Drive'.

**Reply by drveresh** [reply]
@manugarg Yes, I tried all combinations, stop/reset/start, and even disable the extension and tried again yet the same issues. I tried both logout and login (oauth) as well 1+ times, but no results, even with export/import that file.

Stopping and re-setting up didn't create a new file, because I could get my old data back, though it should not delete it, but re-use and probably recreate a new version of file based on the extension version check. I still can't sync.

**Reply by manugarg** [reply]
Sorry to hear that. I am not sure what's going on with your setup, but I'll suggest trying the following steps:

1. **Important**: Take the backup of your notes, either from the `options` page or from the 'All Notes' page. You may need this backup to restore notes from if something doesn't work out.
2. On the `options` page, disable sync. Click on the 'Stop Sync' button. Do this on all machines that you use. 
3. Go to drive.google.com. Search for the `pagenotes.data` file and delete it (move it to trash).
4. On the `options` page, setup sync again.

It will create a new pagenotes.data file in your Google drive and sync your local notes to this new file. Ideally you should not see any data loss or need to recover notes from the backup.

**Reply by medontcode** [reply]
Hi Manu,

I'm having issues with Sync. I have deleted the pagenotes.data in Google Drive, uninstalled and reinstalled Page Notes.

When I click on Setup Sync.
I get "Error: The user turned off browser signin" once.

And if I try again I get "sync: No Oauth token found." 

There is no pagenotes.data being created in my Google Drive.

I have uninstalled and reinstalled Page Notes multiple times and the issue persists.

Help please. Thanks.


===============

Never mind. Figured it out.
I enabled "Allow Chrome sign-in" in Settings and it's working now.

**Reply by manugarg** [reply]
@medontcode Glad that you found the solution. I was not aware of this requirement -- much thanks for reporting this.
